server/src/main/java/org/candlepin/model/Content.java

```diff
@@ -148,6 +172,103 @@ public Content(String id) {
+     * <p/>
+     * Unlike the merge method, all properties from the source content are copied, including the
+     * state of any null collections and any identifier fields.
+     *
+     * @param source
+     *  The Content instance to copy
+     */
+    protected Content(Content source) {
+        this.setUuid(source.getUuid());
+        this.setId(source.getId());
+
+        this.setCreated(source.getCreated());
+        this.setUpdated(source.getUpdated());
+
+        this.setType(source.getType());
+        this.setLabel(source.getLabel());
+        this.setOwners(source.getOwners());
+        this.setName(source.getName());
+        this.setVendor(source.getVendor());
+        this.setContentUrl(source.getContentUrl());
+        this.setRequiredTags(source.getRequiredTags());
+        this.setReleaseVer(source.getReleaseVer());
+        this.setGpgUrl(source.getGpgUrl());
+        this.setMetadataExpire(source.getMetadataExpire());
+        this.setModifiedProductIds(source.getModifiedProductIds());
+        this.setArches(source.getArches());
+    }
+
+    /**
+     * Copies several properties from the given content on to this content instance. Properties that
+     * are not copied over include any identifiying fields (UUID, ID), the creation date and locking
+     * states. Values on the source content which are null will be ignored.
+     *
+     * @param source
+     *  The source content instance from which to pull content information
+     *
+     * @return
+     *  this content instance
+     */
+    public Content merge(Content source) {
+        this.setUpdated(source.getUpdated());
+
+        if (source.getType() != null) {
+            this.setType(source.getType());
+        }
+
+        if (source.getLabel() != null) {
+            this.setLabel(source.getLabel());
+        }
+
+        if (source.getName() != null) {
+            this.setName(source.getName());
+        }
+
+        if (source.getVendor() != null) {
+            this.setVendor(source.getVendor());
+        }
+
+        if (source.getContentUrl() != null) {
+            this.setContentUrl(source.getContentUrl());
+        }
+
+        if (source.getRequiredTags() != null) {
+            this.setRequiredTags(source.getRequiredTags());
+        }
+
+        if (source.getReleaseVer() != null) {
+            this.setReleaseVer(source.getReleaseVer());
+        }
+
+        if (source.getGpgUrl() != null) {
+            this.setGpgUrl(source.getGpgUrl());
+        }
+
+        if (source.getMetadataExpire() != null) {
+            this.setMetadataExpire(source.getMetadataExpire());
+        }
+
+        if (source.getModifiedProductIds() != null) {
+            this.setModifiedProductIds(source.getModifiedProductIds());
+        }
+
+        if (source.getArches() != null) {
+            this.setArches(source.getArches());
+        }
+
+        return this;
+    }
+
+    @Override
+    public Object clone() {
+        return new Content(this);
```

Comment:
`clone` is pretty tricky.  I'm looking at Josh Bloch's *Effective Java* and he's got a [long section](http://jtechies.blogspot.com/2012/07/item-11-override-clone-judiciously.html) on it with a bunch of warnings and comments.  Ultimately he advises just skipping `clone` if at all possible and either using a copy constructor which you wrote or a copy factory `public static Content newInstance(Content toCopy)`.